crates/core/tests/core-wasi-test/src/imagenet.rs

```diff
@@ -0,0 +1,160 @@
+
+
+use crate::ml::test::test::{graph, tensor, inference};
+use image2tensor;
+use image2tensor::convert_image_to_tensor_bytes;
+
+use crate::Path;
+use crate::imagenet_classes;
+
+pub fn elapsed_to_string(fn_name: &str, elapsed: u128) -> String {
+    if elapsed < 1000 {
+        format!("`{}` took {} ns", fn_name, elapsed)
+    } else if elapsed < 1000 * 1000 {
+        format!("`{}` took {:.2} µs", fn_name, elapsed as f64 / 1000.0)
+    } else {
+        format!("`{}` took {:.2} ms", fn_name, elapsed as f64 / 1000.0 / 1000.0)
+    }
+}
+
+pub fn bytes_to_string(b: usize) -> String {
+    if b < 1024 {
+        format!("{} Bytes", b)
+    } else if b < 1024 * 1024 {
+        format!("{:.2} kB", b as f64 / 1024.0)
+    } else {
+        format!("{:.2} MB", b as f64 / 1024.0 / 1024.0)
+    }
+}
+
+/// Return the execution target type from string
+fn map_string_to_execution_target(target: &str) -> Result<graph::ExecutionTarget, String> {
+    match target {
+        "CPU" => Ok(graph::ExecutionTarget::Cpu),
+        "GPU" => Ok(graph::ExecutionTarget::Gpu),
+        "TPU" => Ok(graph::ExecutionTarget::Tpu),
+        _ => {
+            Err(format!("Unknown execution targer = {}", target))
+        }
+    }
+}
+
+
+pub fn imagenet_openvino_test(path_as_string: String, target_as_string: String, image_file: String) -> std::result::Result<(), Box<dyn std::error::Error>> {
+    let path = Path::new(&path_as_string);
+    let target = map_string_to_execution_target(&target_as_string)?;
+    let model = {
+        let start_for_elapsed_macro = std::time::Instant::now();
+        let model: Vec<u8> = std::fs::read(&path.join("model.xml"))?;
+        let elapsed = start_for_elapsed_macro.elapsed().as_nanos();
+        eprintln!("Loaded model from xml {} {}", bytes_to_string(model.len()), elapsed_to_string("fs::read", elapsed));
```

Comment:
If you keep the "elapsed" as Duration instead of converting as_nanos() then in elapsed_to_string() you can use as_millis(), as_micros(), as_nanos(). Just a nitty observation and not a suggestion.

Reply:
implemented